log/trace/README.md

```diff
@@ -0,0 +1,55 @@
+# Trace
+
+This package provides interfaces for tracing functions and printing traces into logs.
+
+## Usage
+
+Inside a function, add these two lines to trace it:
+
+```
+func DoWork(ctx context.Context) err error {
+	ctx, span := StartSpan(ctx, "<span.context>")
+	defer func() { EndSpan(ctx, span, err) }()
```

Comment:
The wrapping function here is not needed:

```suggestion
	defer EndSpan(ctx, span, err)
```

https://go.dev/play/p/ebchS4IUmP8

```go
// You can edit this code!
// Click here and start typing.
package main

import "fmt"

func pr(a string) {
	fmt.Println("pr", a)
}

func toDefer() {
	defer pr("toDefer()")
}

func main() {
	fmt.Println("pre-defer")
	toDefer()
	defer pr("defer pr()")
	fmt.Println("post-defer")
}
```

output:

```
pre-defer
pr toDefer()
post-defer
pr defer pr()
```

Reply:
> The wrapping function here is not needed:

If removing the wrapping function, the `err` will be captured as `nil`.

```
package main

import (
	"fmt"
	"errors"
)

func pr(l string, err error) {
	fmt.Printf("[%s] err:%v\n", l, err)
}

func f() (err error) {
	defer pr("no-wrapper", err)
	defer func() { pr("with-wrapper", err)}()

	return errors.New("An error")
}

func main() {
	f()
}
```
[with-wrapper] err:An error
[no-wrapper] err:<nil>
```

Reply:
🤦 Right, sorry!